examples/src/example-amqp-output.js

```diff
@@ -0,0 +1,151 @@
+
+// Bring in Package Dependencies
+const { default: RulesHarvester, CoreOutputAmqp } = require('rule-harvester');
+const AmqpCacoon = require('amqp-cacoon').default;
+
+// Bring in other Application Specific dependencies
+const RuleInputProviderDirectoryWatcher = require('./providers/rule_input_directory_watcher');
+const ruleClosures = require('./providers/rule_closures');
+const ruleCorpus = require('./providers/rule_corpus');
+const logger = require('./providers/custom_logger');
+
+// Bring in our AMQP Broker configuration
+const amqpConfig = require('./conf/amqp-config');
+
+// Since the AMQP Output requires an AMQP Cacoon object, let's start by creating that.
+// AMQP Cacoon is a library that makes it easy to connect to RabbitMQ.
+let amqpCacoon = new AmqpCacoon({
+  protocol: amqpConfig.protocol,
+  username: amqpConfig.username,
+  password: amqpConfig.password,
+  host: amqpConfig.host,
+  port: amqpConfig.port,
+  amqp_opts: amqpConfig.amqp_opts,
+  providers: {
+    logger: logger,
+  },
+  // Important - onChannelConnect will ensure a certain configuration exists in RMQ.
+  // This might not be needed in environments where RMQ is setup by some other process!
+  onChannelConnect: async (channel) => {
+    try {
+      // Notice all of these are done in sequence with AWAIT. This is so that each
+      // operation can depend on the prior operation having finished. This is important
+      // when binding Queues to Exchanges, for example because you need both the
+      // Exchange and Queue to exist prior to trying to bind them together.
+
+      // Make sure we have our example queue
+      await channel.assertQueue(amqpConfig.exampleQueueOutput, {
+        autoDelete: true,
+        durable: false,
+      });
+      // Make sure we have another example queue for messages with routing keys
+      await channel.assertQueue(
+        `${amqpConfig.exampleQueueOutput}-online-orders`,
+        {
+          autoDelete: true,
+          durable: false,
+        }
+      );
+      // Make sure we have our example exchange
+      await channel.assertExchange(amqpConfig.exampleExchangeOutput, 'direct', {
+        autoDelete: true,
+        durable: false,
+      });
+      // Bind the new Exchange and Queue together
+      await channel.bindQueue(
+        amqpConfig.exampleQueueOutput,
+        amqpConfig.exampleExchangeOutput,
+        '' // Empty routing key to match anything published without one! (Messages published into this
+        // exchange without a routing key WILL be sent to the bound queue.
+      );
+      // Bind the new Exchange and Queue together, but for the second queue with routing key
+      await channel.bindQueue(
+        `${amqpConfig.exampleQueueOutput}-online-orders`,
+        amqpConfig.exampleExchangeOutput,
+        'online-orders' // Empty routing key to match anything published without one! (Messages published into this
+        // exchange without a routing key WILL be sent to the bound queue.
```

Comment:
This comment should be deleted, as it pertains to the `channel.bindQueue()` block above with the empty routing key.